src/hotspot/cpu/s390/s390.ad

```diff
@@ -10273,14 +10273,13 @@ instruct string_inflate_const(Universe dummy, iRegP src, iRegP dst, iRegI tmp, i
 %}
 
 // StringCoding.java intrinsics
-instruct has_negatives(rarg5RegP ary1, iRegI len, iRegI result, roddRegI oddReg, revenRegI evenReg, iRegI tmp, flagsReg cr) %{
-  match(Set result (HasNegatives ary1 len));
-  effect(TEMP_DEF result, USE_KILL ary1, TEMP oddReg, TEMP evenReg, TEMP tmp, KILL cr); // R0, R1 are killed, too.
+instruct count_positives(rarg5RegP ary1, iRegI len, iRegI result, iRegI tmp, flagsReg cr) %{
+  match(Set result (CountPositives ary1 len));
+  effect(TEMP_DEF result, USE_KILL ary1, TEMP tmp, KILL cr); // R0, R1 are killed, too.
```

Comment:
You preserve ary1 in the new implementation. So, you can use iRegP ary1 and remove the USE_KILL effect.

Reply:
Can I be sure then that ary1 isn't allocated the same register as result. Does the effect TEMP_DEF result prevent that? I vaguely remember there was an issue with the initial implementation of has_negatives.

Reply:
Yes, TEMP(_DEF) and USE registers are disjoint.